### PR TITLE
[MPP-1714] Popup Refactor - Settings Panel

### PR DESCRIPTION
**Pull request by maxxcrawford**
Related: https://github.com/mozilla/fx-private-relay-add-on/pull/453

## Summary

Add "Settings" panel to add-on toolbar

Caveat: The "Report Issue" panel is not completed and will be added in a future PR. 

Links: 
- Jira Epic: https://mozilla-hub.atlassian.net/browse/MPP-1708
- [Figma designs](https://www.figma.com/file/NP5tLeIFJeHC5HvaqpLQdp/MPPUX-129-Add-On?node-id=144%3A5071&t=MN61ZFZ9w2429nNj-4) 

Work completed so far: 
- Style "Settings" panel
- Stub "Report Issue" panel

TODO: 
- [x] Add logic refactor for parent/child panel item for panel updates
- [x] Add click event for Mozilla Interaction Data

## Tests

1. Show Settings when logged out
- Run add-on
- Do not log in
- Open pop-up
- Expected: Only two links should be visible: Manage and Settings (Stats and News are hidden)

2. All settings links work
- Run add-on
- Log in
- Open pop-up > Click Gear icon
- Expected: Settings panel opens
- Click Toggles 
- Expected: Toggles should flip to grey/inactive 
- Click links
- Expected: Links should open in new tab (and popup should close) 

3. Report Issue (webcompat) panel should open
- Run add-on 
- Log-in 
- Open pop-up > Click Gear icon
- Click "Report Issue"
- Expected: Report Issue panel should open, and the gear icon should stay "active"
- Click back button
- Expected: Settings panel should re-open, and the gear icon should stay "active"

## Screenshots

Logged in:
image

Logged out:
image

### CODE REVIEW

File: src/css/popup-new.css [hunk 1]
```diff
@@ -180,3 +186,112 @@ sign-up-panel::after {
...
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  width: 100%;
+}
+
+.fx-relay-settings-toggle-wrapper-spacer {
+  height: 1px;
+  background-color: var(--colorGrey10);
+}
```
Comment (Vinnl) [inline src/css/popup-new.css:210]: Woah, a spacer? Table layout flashbacks :sweat_smile: Any reason this couldn't be a padding, gap or border? (Preferably in a comment :) )

File: src/css/popup-new.css [hunk 2]
```diff
@@ -180,3 +186,112 @@ sign-up-panel::after {
...
+  bottom: 0;
+  margin: auto;
+  right: 4px;
+  transition: all 0.2s ease;
+}
+
+.data-disabled::after,
+.input-icons-disabled::after {
+  left: 4px;
+  right: 18px;
```
Comment (Vinnl) [inline src/css/popup-new.css:257]: These magic numbers seem suspicious? Might warrant a comment or other layout technique.

File: src/js/popup/popup-new.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+/* global */
+
```
Comment (Vinnl) [inline src/js/popup/popup-new.js:2]: This doesn't mention any globals?

```suggestion
```

**Review by Vinnl — APPROVED** [review]
Took a pretty superficial look, noted a couple of things that jumped out to me.

File: src/css/popup-new.css (hunk 2 above)
Comment (maxxcrawford) [reply src/css/popup-new.css:257]: Extremely. This was ported over directly from [relay.css](https://github.com/mozilla/fx-private-relay-add-on/blob/main/src/css/relay.css#L225-L230)

File: src/js/popup/popup-new.js (hunk 3 above)
Comment (maxxcrawford) [reply src/js/popup/popup-new.js]: Gahhh. Commented code noise. Good call out. Removed in a82836bffbe352126a8831b55a88a9b9c26a89da

File: src/css/popup-new.css (hunk 1 above)
Comment (maxxcrawford) [reply src/css/popup-new.css:210]: image

So my thought here was not to put padding on the child items (pink/green outlines), but rather use flexbox/gap on the parent. 

The alternate would be to drop the gap from the parent, and add `border-top` to one of the child items (dropping the spacer all together).

Comment (maxxcrawford) [reply src/css/popup-new.css]: Alright. On reflection, having unused markup HTML for CSS sake is bad practice.

Revised in 8b6669dabdd24d6443a837ac9a1288a3b2356f97

Preview: 
image

**Pull request merged**